docs/CODE_GUIDELINES.md

```diff
@@ -341,28 +358,49 @@ int const * const someIntPointer;
 
 ### 5.4. Initialization
 
-Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor to zero-initialize with `{}`.
 
 ✅ Good:
 ```cpp
-int x{};
-int* y{};
-CLog::Log("test: {} {}", x, y);
+int x{3};
+CLog::Log("test: {}", x); // variable used just after its declaration
 ```
 
 ❌ Bad:
 ```cpp
-int x; // used uninitialized
-int* y = nullptr; // default-initialization not using {}
-CLog::Log("test: {} {}", x, y);
+int x{3}; // variable not immediately used, if possible declare the variable close to the point of its use 
+[...many lines of code that do not use variable x...]
+CLog::Log("test: {}", x);
 ```
 
-In general, prefer the `{}` initializer syntax over alternatives. This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
+Although we suggest adopting the use of `{}` for the local variables, like the example above,
+for this use case we allow to initialize the variables also by using the form `=` to improve the code readability.
 
 ✅ Good:
 ```cpp
-int x{5};
-int y{x};
+std::string localVariableX{m_fooVariableOne + m_barVariableTwo}; // less readable
+std::string localVariableX = m_fooVariableOne + m_barVariableTwo; // more readable
+```
+
+In any case, the form `{}` must always be used for members of classes and structures.
```

Comment:
why is this different from local variables?

Reply:
has been widely discussed in the past on slack, on differents PRs, maybe you can find also some old comments above,
it is only for ease of reading when the code blocks are broad,
if you want to reopen this topic, open a discussion on Slack and get everyone involved, for now this was the last decision made

Reply:
I'll ask on slack, thanks

Reply:
in an ideal case, though, the code guidelines should allow for understanding their reasoning where rules go beyond simple matters of choice ;)